pkg/meta/boltdb.go

```diff
@@ -0,0 +1,140 @@
+package meta
+
+import (
+	"os"
+	"path"
+	"time"
+
+	boltdb "github.com/boltdb/bolt"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	Register("boltdb", &bolt{})
+}
+
+type bolt struct {
+	db *boltdb.DB
+}
+
+// New is used to make bolt metadata store instance.
+func (b *bolt) New(cfg Config) error {
+	opt := &boltdb.Options{
+		Timeout: time.Second * 10,
+	}
+
+	if _, err := os.Stat(cfg.BaseDir); err != nil && os.IsNotExist(err) {
+		if err := os.MkdirAll(path.Dir(cfg.BaseDir), 0755); err != nil {
+			return errors.Wrap(err, "failed to create metadata path")
+		}
+	}
+
+	db, err := boltdb.Open(cfg.BaseDir, 0644, opt)
+	if err != nil {
+		return err
+	}
+	for _, bucket := range cfg.Buckets {
+		if err := b.prepare(db, []byte(bucket.Name)); err != nil {
+			return err
+		}
+	}
+	b.db = db
+
+	return nil
+}
+
+func (b *bolt) prepare(db *boltdb.DB, bucket []byte) error {
+	return db.Update(func(tx *boltdb.Tx) error {
+		_, err := tx.CreateBucketIfNotExists(bucket)
+		if err != nil {
+			return errors.Wrap(err, "boltdb: create bucket")
```

Comment:
If that, the error string would be `boltdb: create bucket: err.Error()`. I am wondering if we could make it like `failed to create bucket in boltdb: err.Error()`, so the format should be `return errors.Wrap(err, "failed to create bucket in boltdb")` @rudyfly

Reply:
ok